lib/syskit/data_service.rb

```diff
@@ -90,10 +90,13 @@ def find_device_attached_to(service = nil)
                     elsif driver_services.size > 1
                         raise ArgumentError, "#{self} handles more than one device, you must specify one of #{driver_services.map(&:name).sort.join(", ")} explicitely"
                     end
-                    service = driver_services.first
+                    service = driver_services.find do |driver_service|
+                        service == driver_service
+                    end
+                    
                 end
 
-                arguments["#{service.name}_dev"]
```

Comment:
> The behaviour should be that a task context can be driver for an arbitrary number of devices of any types.
The behavior after the PR is the following: A TaskContext can be declared as being a driver for a numer of different devices:
```
JointControl::FakeJointDriverTask.driver_for BesmanTest::FakeDevices::JointsPositionDriver, :as => 'dummy_position'
JointControl::FakeJointDriverTask.driver_for BesmanTest::FakeDevices::JointsVelocityDriver, :as => 'dummy_velocity'
```

But then, a concrete TaskContext, once instanciated can only be either ```BesmanTest::FakeDevices::JointsVelocityDriver``` or ```BesmanTest::FakeDevices::JointsPositionDriver```. 

There is no problem, where there are two concrete TaskContext ```JointControl::FakeJointDriverTask```, representing different devices:
```
            device(BesmanTest::FakeDevices::JointsPositionDriver, :as => 'right_hand') \
                .joint_names(*$right_hand_joints) \
                .with_conf('right_hand', 'position').prefer_deployed_tasks(/right_hand/)
            device(BesmanTest::FakeDevices::JointsVelocityDriver, :as => 'torso') \
                .joint_names(*$torso_joints)\
                .with_conf('torso', 'velocity').prefer_deployed_tasks(/torso/)
```
for example is valid.

Does this example help?